swim/bootstrap_test.go

```diff
@@ -88,6 +88,22 @@ func (s *BootstrapTestSuite) TestBootstrapDestroy() {
 	s.Error(lerr.Err(), "expected bootstrap to fail if node gets destroyed")
 }
 
+func (s *BootstrapTestSuite) TestJoinHandlerNotMakingAlive() {
+	// get a bootstrapped cluster
+	s.peers = genChannelNodes(s.T(), 3)
+	bootstrapList := bootstrapNodes(s.T(), true, s.peers...)
+
+	s.tnode.node.Bootstrap(&BootstrapOptions{
+		Hosts: bootstrapList,
+	})
+
+	// test that there are no changes to disseminate from the peers after bootstrapping the t(est)node
+	for _, peer := range s.peers {
```

Comment:
A more accurate assertion or test would be to verify that when calling the /protocol/join endpoint, the caller (joiner) is not added to membership. Testing whether the disseminator has changes is a roundabout (and potentially frail) way of verifying that the member was not made alive. It relies too much on the interactions between the membership component and the disseminator.